lib/Sema/TypeCheckEffects.cpp

```diff
@@ -2477,15 +2477,28 @@ class CheckEffectsCoverage : public EffectsHandlingWalker<CheckEffectsCoverage>
         effects.push_back(EffectKind::Async);
       }
 
-      checkThrowAsyncSite(E, getter->hasThrows(),
+      bool requiresTry = getter->hasThrows();
+      checkThrowAsyncSite(E, requiresTry,
                           Classification::forEffect(effects,
                                   ConditionalEffectKind::Always,
                                   getKindOfEffectfulProp(member)));
 
-    } else if (E->isImplicitlyAsync()) {
-      checkThrowAsyncSite(E, /*requiresTry=*/false,
-            Classification::forUnconditional(EffectKind::Async,
-                                             getKindOfEffectfulProp(member)));
+    } else {
```

Comment:
@kavon this refactoring re effect handling is what broke the properties, please have a look if this looks good.

I figured a general "else" sounds better than special handling just the implicit async and inside the handling the implicit throws.